File: cpp/p3.cpp
class Solution {
public:
ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {

    ListNode *dummy = new ListNode(-1), *pre = dummy, *new_start = nullptr;
    int out = 0, carry = 0;
    dummy -> next = l1; 
    
    
    while(l1 != NULL && l2 != NULL){
        // insert codehere
        
        
    }
    
   
    new_start = (l1 == NULL) ? l2: l1;        
    pre -> next = new_start;
    
    while(new_start != NULL){
        pre = new_start;
        pre = (new_start -> val + carry)/10;
        new_start -> val = out;
        
        new_start = new_start -> next;
    }
    
    //  determine the last node;
    if(carry != 0){
       // insert code here
    }
    
    return dummy -> next;
    
}

};

File: java/p3.java
public class p3 {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        
        if(l1 == null ){
           return l2;
        }
       if (l2 == null){
           return l1;
       }
           
       ListNode head = new ListNode(0);
       ListNode header = head;     
       int carry = 0;
           
       while(l1 != null || l2 != null){
           
           if (l1 == null){
               //insert code here
           }else if (l2 == null){
               //insert code here
           }
           
               header.next = new ListNode((l1.val + l2.val + carry) %10);
       
               carry = (l1.val + carry) /10;
       
               header = header.next;
               l1 = l1.next;
               l2 = l2.next; 
           }
           
   
           if (carry != 0){
               header.next = new ListNode(carry);
           }
      
           return head.next; 
           
       
       }
    
}
